### PR TITLE
WFLY-9703 New archetype for EAR project for WildFly 16

### PR DESCRIPTION
**Pull request by WolfgangHG**
This archetype shall replace the "wildfly-javaee7-webapp-ear-blank-archetype" archetype, which is outdated and can no longer be build because of conceptual changes to the kitchensink-ear quickstart.

If it is committed, someone has to release if to maven central. See the scripts in the wildfly-archetypes root. Also, the old archetype can be removed.

### CODE REVIEW

File: wildfly-javaee-ear-archetype/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    <build>
+        <plugins>
+            <!-- The WildFly plugin deploys your ear to a local JBoss AS container -->
+            <!-- Due to Maven's lack of intelligence with EARs we need 
+                 to configure the wildfly maven plugin to skip deployment for all modules.
+                 We then enable it specifically in the ear module. -->
+            <plugin>
+                <groupId>org.wildfly.plugins</groupId>
+                <artifactId>wildfly-maven-plugin</artifactId>
+				<version>${version.wildfly.maven.plugin}</version>
```
Comment (emmartins) [inline wildfly-javaee-ear-archetype/src/main/resources/archetype-resources/pom.xml]: please fix indention

File: wildfly-javaee-ear-archetype/src/main/resources/archetype-resources/README.txt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+To deploy it:
+Run the maven goals "install wildfly:deploy"
+
+To undeploy it:
+Run the maven goals "wildfly:undeploy"
+
+==========================
+
+DataSource:
```
[the remark this answers is not in the document]
Comment (WolfgangHG) [reply wildfly-javaee-ear-archetype/src/main/resources/archetype-resources/README.txt]: done

File: wildfly-javaee-ear-archetype/src/main/resources/archetype-resources/__rootArtifactId__-ejb/src/main/resources/META-INF/beans.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (WolfgangHG) [reply wildfly-javaee-ear-archetype/src/main/resources/archetype-resources/__rootArtifactId__-ejb/src/main/resources/META-INF/beans.xml]: removed

File: wildfly-javaee-ear-archetype/src/main/resources/archetype-resources/__rootArtifactId__-ejb/src/main/resources/META-INF/persistence.xml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        http://xmlns.jcp.org/xml/ns/persistence/persistence_2_1.xsd">
+   <persistence-unit name="primary">
+      <!-- If you are running in a production environment, add a managed
+         data source, this example data source is just for development and testing! -->
+      <!-- The datasource is deployed as <EAR>/META-INF/${rootArtifactId}-ds.xml, you
+         can find it in the source at ear/src/main/application/META-INF/${rootArtifactId}-ds.xml -->
+      <jta-data-source>java:jboss/datasources/${rootArtifactId}DS</jta-data-source>
+      <properties>
+         <!-- Properties for Hibernate -->
+         <property name="hibernate.hbm2ddl.auto" value="create-drop" />
```
[the remark this answers is not in the document]
Comment (WolfgangHG) [reply wildfly-javaee-ear-archetype/src/main/resources/archetype-resources/__rootArtifactId__-ejb/src/main/resources/META-INF/persistence.xml]: done, but added a comment that the tables won't be removed on undeploy - and a hint to the "old" property.

File: wildfly-javaee-ear-archetype/src/main/resources/archetype-resources/__rootArtifactId__-ear/src/main/application/META-INF/__rootArtifactId__-ds.xml [hunk 5]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (WolfgangHG) [reply wildfly-javaee-ear-archetype/src/main/resources/archetype-resources/__rootArtifactId__-ear/src/main/application/META-INF/__rootArtifactId__-ds.xml]: deleted

File: wildfly-javaee-ear-archetype/src/main/resources/archetype-resources/__rootArtifactId__-web/src/test/java/test/SampleIT.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+        // Now grab the web archive:
+        f = new File("./target/${rootArtifactId}-web.war");
+        if (f.exists() == false) {
+            throw new RuntimeException("File " + f.getAbsolutePath() + " does not exist.");
+        }
+        WebArchive war = ShrinkWrap.create(ZipImporter.class, "${rootArtifactId}-web.war").importFrom(f).as(WebArchive.class);
+        ear.addAsModule(war);
+
+        // Add the datasource definition:
```
[the remark this answers is not in the document]
Comment (WolfgangHG) [reply wildfly-javaee-ear-archetype/src/main/resources/archetype-resources/__rootArtifactId__-web/src/test/java/test/SampleIT.java]: done

File: wildfly-javaee-ear-archetype/README.adoc [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+----
+$ mvn clean install
+----
+
+[[newwildflyversion]]
+==== New WildFly version
+To update this archetype to a new WildFly version, just update the version properties in "src/main/resources/archetype-resources/pom.xml".
+Check whether dependencies have changed.
+
+Also test whether the Arquillian unit test "SampleIT" still works. Test it using both profiles "arq-remote" and "arq-managed".
```
[the remark this answers is not in the document]
Comment (WolfgangHG) [reply wildfly-javaee-ear-archetype/README.adoc]: Is it better now ;-)?

File: wildfly-javaee-ear-archetype/src/main/resources/archetype-resources/__rootArtifactId__-web/src/main/webapp/WEB-INF/beans.xml [hunk 8]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (WolfgangHG) [reply wildfly-javaee-ear-archetype/src/main/resources/archetype-resources/__rootArtifactId__-web/src/main/webapp/WEB-INF/beans.xml]: removed

File: wildfly-javaee-ear-archetype/src/main/resources/archetype-resources/pom.xml (hunk 1 above)
Comment (WolfgangHG) [reply wildfly-javaee-ear-archetype/src/main/resources/archetype-resources/pom.xml]: done. Damn editors...

**Comment by emmartins** [comment]
@WolfgangHG I have further optimised/fixed the PR a bit, and issued a new PR at #4

**Pull request merged**